server_vllm.py

```diff
@@ -304,12 +304,24 @@ async def abort_request() -> None:
         await engine.abort(request_id)
 
     async def wrap_vllm_generator(
-        tool_choice,
+        tool_choice, is_tools
     ) -> AsyncGenerator[Tuple[str, Optional[str]], None]:
         previous_texts = ""
+        tool_call_count = 0
         async for res in result_generator:
+            function_call_break = False
             for output in res.outputs:
                 delta_text = output.text[len(previous_texts) :]
+                if not is_tools:
+                    stop_stream, tool_call_count = (
```

Comment:
Each prompt template has a different way of tracking. For example. v2 requires the count to be 1 while v2.llama3 requires the count to be 2. Thus, I create version-specific `check_function_call_to_stop_streaming` method.

Reply:
But we can base on the result from streaming to count, right ? because when it is the start of function_call the first delta would contain the function_name but later delta function_name=NOne and arguments != None